sync-icloud.sh

```diff
@@ -978,9 +978,9 @@ Notify(){
          -F "message=${notification_message}")"   
    elif [ "${notification_type}" = "Bark" ]; then
       if [ "${notification_files_preview_count}" ]; then
-         bark_text="$(echo -e "${notification_icon} \n${notification_message}\nMost recent ${notification_files_preview_count} ${notification_files_preview_type} files:\n${notification_files_preview_text//_/\\_}")"
+         bark_text="$(echo -e "${notification_icon} ${notification_message} Most recent ${notification_files_preview_count} ${notification_files_preview_type} files: ${notification_files_preview_text//_/\\_}")"
```

Comment:
I'm not quite sure what to do with this line ending (`${notification_files_preview_text//_/\\_}`).

Reply:
> I'm not quite sure what to do with this line ending (`${notification_files_preview_text//_/\\_}`).

This is due to Telegram needing underscore characters to be escaped or it causes issues with the API. If the message look OK, we'll leave it as it is now.

Reply:
I tested with the latest version.

Finally, I was able to receive a notification when the sync started.

However, when the notification is sent after the synchronization is completed, it still fails. I think the notification fails because there are unsupported escape characters in the notification content.

```
2022-10-11 10:00:24     INFO    111.111.111.111 -> [200] POST   186ms /push => /push { "device_key": "XXX", "title": "boredazfcuk/iCloudPD", "body": "▶ 启动成功，开始同步当前 Apple ID 中的照片", "category": "category" }

2022-10-11 10:21:44     INFO    111.111.111.111 -> [400] POST     6ms /push => /push { "device_key": "XXX", "title": "boredazfcuk/iCloudPD", "body": "⏬ iCloud 图库同步完成，新增 5 张照片 Most recent 5 下载 files: 2022/10/11/incoming-710FEAF4-5327-4755-8CB3-E1F4DD76F196.PNG
```

Probably due to newlines between photo filenames. Causes the JSON data in the request to be incomplete and ends at the end of the first photo filename.

Hope you can help me solve this problem, thanks a lot.

Reply:
It seems that there is something not quite right with the formatting. There are many JSON based notification messages which work with new line characters, I'm not sure why this one isn't.

In the original issue, you mentioned testing the notifications from the command line with:
```
curl --silent --output /dev/null --write-out "%{http_code}" "http://api.day.app/push" \
         -H 'Content-Type: application/json; charset=utf-8' \
         -d "{ \"device_key\": \"XXX\", \"title\": \"Hello\", \"body\": \"World\", \"category\": \"category\" }"
```
Can you do some more testing to see what escaping is needed to correctly pass a newline character. e.g.:

```
curl --silent --output /dev/null --write-out "%{http_code}" "http://api.day.app/push" \
         -H 'Content-Type: application/json; charset=utf-8' \
         -d "{ \"device_key\": \"XXX\", \"title\": \"Hello\", \"body\": \"Hello\nWorld\", \"category\": \"category\" }"
```
```
curl --silent --output /dev/null --write-out "%{http_code}" "http://api.day.app/push" \
         -H 'Content-Type: application/json; charset=utf-8' \
         -d "{ \"device_key\": \"XXX\", \"title\": \"Hello\", \"body\": \"Hello\\nWorld\", \"category\": \"category\" }"
```

This would suggest that the fix to the original issue would have been to change this:
```
bark_text="$(echo -e "${notification_icon} \n${notification_message}\nMost recent ${notification_files_preview_count} ${notification_files_preview_type} files:\n${notification_files_preview_text//_/\\_}")"
```
to this:
```
bark_text="$(echo -e "${notification_icon} \\n${notification_message}\\nMost recent ${notification_files_preview_count} ${notification_files_preview_type} files:\\n${notification_files_preview_text//_/\\_}")"
```
rather than removing them entirely.

Reply:
In both tests you gave, I can get notifications and both line breaks correctly.

Reply:
```
/ # curl --silent --output /dev/null --write-out "%{http_code}" "http://api.day.app/push" \
>          -H 'Content-Type: application/json; charset=utf-8' \
>          -d "{ \"device_key\": \"xxx\", \"title\": \"Hello\", \"body\": \"Hello\\n
World\", \"category\": \"category\" }"
200/ # echo "$?"
0
/ # 
```

```
/ # curl --silent --output /dev/null --write-out "%{http_code}" "http://api.day.app/push" \
>          -H 'Content-Type: application/json; charset=utf-8' \
>          -d "{ \"device_key\": \"xxx\", \"title\": \"Hello\", \"body\": \"He
llo\\nWorld\", \"category\": \"category\" }"
200/ # echo "$?"
0
/ # 
```

exit codes are all "0".

Can you provide me with some actual notification content to test? Instead of simple notification content like `\"Hello\nWorld\"`.